props/saber_caiwyn_buttons.h

```diff
@@ -369,24 +368,22 @@ class CaiwynButtons : public PropBase {
   }
 
   void NextTrack(bool manually_selected) {
-    if (manually_selected) {
-      track_player_->Stop();
-      track_player_.Free();
-#ifdef CAIWYN_SAVE_TRACKS
-      current_preset_.track = mkstr(current_track_);
-      current_preset_.Save();
-#endif
-      if (SFX_mselect) {
-        hybrid_font.PlayPolyphonic(&SFX_mselect);
-      }
-    }
+    if (track_player_->isPlaying()) track_player_->Stop();
```

Comment:
Btw, this will crash if track_player_ is null.

Reply:
That would be the reason, then.  NextTrack() can be called by a button press, or it can be called when a track finishes playing and the track player mode is PLAYBACK_ROTATE.  But PropBase::Loop() frees the track player when a track finishes, so I have to call GetWavPlayer() to set it back up again.  NextTrack() stops and frees the track player if it's currently playing in anticipation of that.

I'm wondering: could I avoid this by not calling PropBase::Loop() from my own Loop() override?  Could I instead just replicate everything in PropBase::Loop() except the code that frees the tracks?  Like so:

  void Loop() override {
    PropBase::CallMotion();
    if (clash_pending1_) {
      clash_pending1_ = false;
      PropBase::Clash(pending_clash_is_stab1_, pending_clash_strength1_);
    }
    if (clash_pending_ && millis() - last_clash_ >= clash_timeout_) {
      clash_pending_ = false;
      PropBase::Clash2(pending_clash_is_stab_, pending_clash_strength_);
    }
    PropBase::PollScanId();
    PropBase::CheckLowBattery();
  }

Reply:
You could, but why?
If you do that you would reserve one of the 7 wav players forever, even when you don't need it.

I think what I would do is to change PlayTrack to do:

if (!track_player_) track_player_ = GetFreeTrackPlayer();

That way, it would re-use the same track player if there already is one, or try to get a new one if there isn't.

Then there would be no reason to free the track player.